rgw/v2/lib/s3/lifecycle_validation.py

```diff
@@ -85,39 +86,120 @@ def validate_prefix_rule(bucket, config):
-        ncurr = 0
-        two_pool_curr_ncurr = 0
-        ec_pool_curr_ncurr = 0
-        for i in range(0, objs_total):
-            storage_class = json_doc2[i]["meta"]["storage_class"]
-            if config.two_pool_transition and storage_class == expected_storage_class:
-                two_pool_curr_ncurr += 1
-            elif config.ec_pool_transition and storage_class == config.ec_storage_class:
-                ec_pool_curr_ncurr += 1
-            else:
-                if storage_class == config.storage_class:
-                    curr += 1
-                else:
-                    ncurr += 1
-        if curr == config.objects_count and ncurr == objs_ncurr:
-            log.info(
-                "Lifecycle transition of current and noncurrent object version validated"
+        if config.test_ops.get("test_cloud_transition", False):
+            log.info("Start the LC cloud transition verification")
+            objs_total = (config.test_ops["version_count"]) * (config.objects_count)
+            log.info(f"total objects are {objs_total} objects.")
+            bucket_stats_op = utils.exec_shell_cmd(
+                f"radosgw-admin bucket stats --bucket={bucket.name}"
             )
-        elif two_pool_curr_ncurr == objs_total:
-            log.info(
-                "Two pool Lifecycle transition of current and noncurrent object version validated."
-            )
-        elif ec_pool_curr_ncurr == objs_total:
-            log.info(
-                "Bucket LC Transition to EC pool of current and noncurrent object version validated."
+            json_doc = json.loads(bucket_stats_op)
+            bucket_list_op = utils.exec_shell_cmd(
+                f"radosgw-admin bucket list --bucket={bucket.name}"
             )
+            json_doc_list = json.loads(bucket_list_op)
+            objects_remaining = json_doc["usage"]["rgw.main"]["num_objects"]
+            count = 0
+            if config.test_ops.get("test_retain_head", False):
+                log.info("Verifying the cloud transition with retain_head_true")
+                if config.test_ops.get("test_cloud_transition_at_remote", False):
+                    zone_name = config.remote_zone
+                    log.info(
+                        "Verifying the cloud transition at remote site with retain_head_true"
+                    )
+                    remote_site_ssh_con = reusable.get_remote_conn_in_multisite()
+                    log.info(
+                        f"collect bucket stats for {bucket.name} at remote site {zone_name}"
+                    )
+                    cmd_bucket_stats = (
+                        f"radosgw-admin bucket stats --bucket {bucket.name}"
+                    )
+                    stdin, stdout, stderr = remote_site_ssh_con.exec_command(
+                        cmd_bucket_stats
+                    )
+                    cmd_output = stdout.read().decode()
+                    stats_remote = json.loads(cmd_output)
```

Comment:
@viduship , could you log status_remote value

JFYI, code alreaady exists for remote_exec_shell_command
https://github.com/red-hat-storage/ceph-qe-scripts/blob/master/rgw/v2/utils/utils.py#L103

Reply:
done.